### PR TITLE
add functionality to choose between 6 digit and 8 character codes

### PR DESCRIPTION
**Pull request by Derpitron**
I defined a function to generate 6 digit code, then gave the user an option to choose between generating a 6 digit or 8 character code

### CODE REVIEW

**Comment by SpaghettDev** [comment]
Hey! Sorry for the late response...

The reason why I didn't add this option is because when you use the SMS codes, which has ~1 million different combinations, which is a high chance of getting it right you might be thinking. The downside is that every time you enter the wrong code, the code sent changes, so you have a less chance of getting it right.
But using the 8 digit backup code, which has ~3 billion different combinations (or something like that), even thought it has that many different combinations, you can just simply not use the codes that you found out to be wrong...
Basically, using the SMS codes is randomized, so if "145 185" is wrong the first time, maybe next time it'll be right! But if "wt68-aq78" (an 8 digit backup code) is wrong, it'll always be wrong.

Nonetheless, it's a nice addition, so I'll test it and then I will merge it.


Thanks for your contribution!
-SpaghettDev

**Comment by SpaghettDev** [comment]
Looks good! Merging.

**Comment by Derpitron** [comment]
Well it's 10^6 possible codes that change every 30 seconds vs 36^8 permanent codes, i'm willing to take chances with 10^6 codes. I originally added the 6 digit code functionality on a private fork of mine for a project i'm doing, then I thought why not commit it upstream?
